x/ibc/transfer/keeper/msg_server.go

```diff
@@ -21,26 +21,18 @@ var _ types.MsgServer = Keeper{}
 // This implementation overrides the default ICS20 transfer's by converting
 // the ERC20 tokens to their Cosmos representation if the token pair has been
 // registered through governance.
+// If user doesnt have enough balance of coin, it will attempt to convert
+// erc20 tokens to the coin denomination, and continue with a regular transfer.
 func (k Keeper) Transfer(goCtx context.Context, msg *types.MsgTransfer) (*types.MsgTransferResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	// check if IBC transfer denom is a valid Ethereum contract address
-	if !common.IsHexAddress(msg.Token.Denom) {
-		// no-op: continue with regular transfer
-		return k.Keeper.Transfer(sdk.WrapSDKContext(ctx), msg)
-	}
-
 	sender, err := sdk.AccAddressFromBech32(msg.Sender)
 	if err != nil {
 		// NOTE: shouldn't happen as the receiving address has already
 		// been validated on ICS20 transfer logic
 		return nil, sdkerrors.Wrap(err, "invalid sender")
 	}
 
-	// Return acknowledgement and continue with the next layer of the IBC middleware
```

Comment:
Undo comment removal

Reply:
This comment was incorrect since we are not on the middleware, no ack to be returned.